tests/test_utils.py

```diff
@@ -118,6 +124,16 @@ def test_get_static_method(path: str, return_value: Any) -> None:
             {"a": 10, "d": 40},
             Bar(10, 200, 200, 40),
         ),
+        (
+            {
+                "cls": "tests.test_utils.Baz",
+                "method": "class_method",
```

Comment:
The syntax I would like to support is:
```yaml
foo:
   cls: tests.test_utils.Baz.class_method
```
Is this implementable?
You may need to differentiate which of the two modes we are in automatically.

Reply:
Yea, I wanted that too, but it didn't seem easy if you use the `instantiate` method because you would have to always check the last and second to last items to see if the last item in the string was a method or a class of the method of the second to last item.  I thought it would create extra logic that made things more confusing.  My other idea was to create a `instantiate_from_class_method` function (or whatever you'd want it to be called) which assumes that form.  Should work either way, it just depends where you want to put the logic.  

I'll take a look at it and see if I can put it all into the `cls` key neatly.

Reply:
Yea, I wanted that too, but it didn't seem easy if you use the `instantiate` method because you would have to always check the last and second to last items to see if the last item in the string was a method or a class of the method of the second to last item.  I thought it would create extra logic that made things more confusing.  My other idea was to create a `instantiate_from_class_method` function (or whatever you'd want it to be called) which assumes that form.  Should work either way, it just depends where you want to put the logic.  

I'll take a look at it and see if I can put it all into the `cls` key neatly.

Reply:
Yeah, the implementation would be more complicated but I think it's okay.
Let's see what it looks like.

By the way, I just realized that PluginConf is now a part of the public API. I originally thought of it as an internal Hydra detail. I will have to promote it to be a part of the API. (probably also rename it).

Reply:
@omry alright, it now works.  I used the the `locate` from from `pydoc` which basically does all the work.  I also simplified the other parts of the code that were duplicating functionality of this such as your static method function.